pkg/diff/chunk_test.go

```diff
@@ -28,121 +32,173 @@ type chunkTestCase struct {
 }
 
 func (*testChunkSuite) TestSplitRange(c *C) {
-	testCases := []*chunkTestCase{
-		{
-			&chunkRange{
-				begin:        int64(1),
-				end:          int64(1000),
-				containBegin: true,
-				containEnd:   true,
-			},
-			1,
-			[]*chunkRange{
-				{
-					begin:        int64(1),
-					end:          int64(1000),
-					containBegin: true,
-					containEnd:   true,
-				},
-				{
-					begin:        struct{}{},
-					end:          int64(1),
-					containBegin: false,
-					containEnd:   false,
-					noBegin:      true,
-				},
-				{
-					begin:        int64(1000),
-					end:          struct{}{},
-					containBegin: false,
-					containEnd:   false,
-					noEnd:        true,
-				},
+	dbConn, err := createConn()
+	c.Assert(err, IsNil)
+
+	_, err = dbConn.Query("CREATE DATABASE IF NOT EXISTS `test`")
+	c.Assert(err, IsNil)
+
+	_, err = dbConn.Query("DROP TABLE IF EXISTS `test`.`testa`")
+	c.Assert(err, IsNil)
+
+	createTableSQL := `CREATE TABLE test.testa (
+		a date NOT NULL,
+		b datetime DEFAULT NULL,
+		c time DEFAULT NULL,
+		d varchar(10) COLLATE latin1_bin DEFAULT NULL,
+		e int(10) DEFAULT NULL,
+		h year(4) DEFAULT NULL,
+		PRIMARY KEY (a))`
+
+	dataCount := 10000
+	cfg := &importer.Config{
+		TableSQL:    createTableSQL,
+		WorkerCount: 5,
+		JobCount:    dataCount,
+		Batch:       100,
+		DBCfg:       dbutil.GetDBConfigFromEnv("test"),
+	}
+
+	// generate data for test.testa
+	importer.DoProcess(cfg)
+	defer dbConn.Query("DROP TABLE IF EXISTS `test`.`testa`")
+
+	// only work on tidb, so don't assert err here
+	_, _ = dbConn.Query("ANALYZE TABLE `test`.`testa`")
```

Comment:
if not TiDB, the code below can worker?

Reply:
will split chunk by random, can still work